databases/week3/week3.sql

```diff
@@ -0,0 +1,166 @@
+DELETE FROM review
+WHERE id = 1;
+
+-- Additional queries
+/*Now add a couple of different meals, reservations and reviews with different attributes. With those meals create the following queries*/
+
+						-- TABLE REVIEW--
+INSERT INTO review (title, description, meal_id, stars, created_date )
+VALUES ('amazing', 'super cervice', '2', '5', '2020-06-01');
+INSERT INTO review (title, description, meal_id, stars, created_date )
+VALUES ('bad', 'too expensive', '5', '1', '2020-06-06');
+INSERT INTO review (title, description, meal_id, stars, created_date )
+VALUES ('great', 'I will come again', '3', '3', '2020-06-05');
+
+						-- TABLE RESERVATION --
+INSERT INTO reservation (number_of_guests, meal_id, created_date)
+VALUES ('2', '3', '2020-06-09');
+INSERT INTO reservation (number_of_guests, meal_id, created_date)
+VALUES ('1', '4', '2020-06-03');
+INSERT INTO reservation (number_of_guests, meal_id, created_date)
+VALUES ('2', '3', '2020-06-07');
+
+						-- TABLE MEAL --
+INSERT INTO meal ( title, description, location, `when`, max_reservation, price, created_date)
+VALUES ('supe', 'with meat bolls', 'south', '2020-06-8', '2', 20.50, '2020-06-04');
+INSERT INTO meal ( title, description, location, `when`, max_reservation, price, created_date)
+VALUES ('spaghetti', 'with  basilic', 'west', '2020-06-10', '3', 45, '2020-06-04');
+INSERT INTO meal ( title, description, location, `when`, max_reservation, price, created_date)
+VALUES ('ice cream', 'organic', 'city center', '2020-06-15', '5', 45, '2020-06-04');
+
+-- Functionality
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT title FROM meal
+WHERE price < 50;
+
+-- Get meals that still has available reservations
+select * from (SELECT meal.title, meal.max_reservation AS max_reservations, COUNT(reservation.id) AS reserved_meals_count
+FROM reservation
+LEFT JOIN meal ON meal.id = reservation.meal_id
+GROUP BY meal.title) AS meals_reservations
+WHERE max_reservations > reserved_meals_count;
+
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT title FROM meal
+WHERE title LIKE 's%';
+
+-- Get meals that has been created between two dates
+SELECT title FROM meal
+WHERE created_date BETWEEN '2020-06-02' AND '2020-06-05';
+
+-- Get only specific number of meals fx return only 5 meals
+SELECT * FROM meal
+LIMIT 2;
+
+-- Get the meals that have good reviews
+SELECT meal.title, review.stars
+FROM review
+INNER JOIN meal ON meal.id = review.meal_id
+WHERE review.stars > '3';
+
+-- Get reservations for a specific meal sorted by created_date
+SELECT meal.title, reservation.created_date
+FROM reservation
+INNER JOIN meal ON meal.id = reservation.meal_id
+ORDER BY reservation.created_date;
+
+-- Sort all meals by average number of stars in the reviews
+SELECT meal.title, review.stars, AVG(review.stars)  AS average
+FROM review
+INNER JOIN meal ON meal.id = review.meal_id
+GROUP BY meal.title
+ORDER BY average;
+
```

Comment:
This one gives me this error: `Error Code: 1055. Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'mealsharing.review.stars' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by`

Can you figure out why?